common/app/templates/inlineJS/blocking/applyAdfreeRenderCondition.scala.js

```diff
@@ -0,0 +1,119 @@
+@()
+
+(function applyAdfreeRenderCondition(commercial){
+    var AD_FREE_COOKIE = 'gu_adfree_test',
+        MVT_ID_COOKIE = 'GU_mvt_id',
+        FIRST_TIME_VISIT_KEY = 'gu.firstVisitTime',
+        ALREADY_VISITED_KEY = 'gu.alreadyVisited';
+
+    if ((readLocalStorage(ALREADY_VISITED_KEY) || 0) == 0) {
+        writeLocalStorage(FIRST_TIME_VISIT_KEY, new Date().getTime());
+    }
+
+    if (isFirst3VisitDays() && isInVariantGroup()){
+        commercial.showingAdfree = true;
+        return 'is-adfree';
+    } else {
+        commercial.showingAdfree = false;
+        return '';
+    }
+
+    function isFirst3VisitDays() {
+        var firstVisit = readLocalStorage('gu.firstVisitTime') || 0;
```

Comment:
You should probably refer to your constant again, here.

Also, when will this read return a null? And in that case, is it appropriate to act as though the user's first visit was at the start of the Unix Epoch?